linodecli/plugins/k8s-alpha.py

```diff
@@ -0,0 +1,286 @@
+    kubeconfig_new = replace_kubeconfig_user(terrapath, parsed.name)
+    kubeconfig_existing = os.path.expanduser("~/.kube/config")
+    # Create a merged kubeconfig file and set the context
+    # First set up the KUBECONFIG env var so that `kubectl config view --flatten`
+    # gives us a new merged config
+    os.environ["KUBECONFIG"] = "{}:{}".format(kubeconfig_existing, kubeconfig_new)
+    tempfilepath = 'tempkubeconfig'
+    tempf = open(tempfilepath, 'w')
+    call_or_exit(['kubectl', 'config', 'view', '--flatten'], stdout=tempf)
+    tempf.close()
+    shutil.move(tempfilepath, kubeconfig_existing)
+
+    # Set the kubeconfig context to the new cluster
+    call_or_exit(['kubectl', 'config', 'use-context', '{}@{}'.format(parsed.name, parsed.name)])
+
+    # We're done! We have merged the user's kubeconfigs.
+    # So, the user should be able to run something like
+    # `kubectl get pods --all-namespaces`
+    # and see the Linode CSI, CCM, and ExternalDNS controllers
+
+def delete(args, context):
+    parser = argparse.ArgumentParser("{} create".format(plugin_name), add_help=True)
+    parser.add_argument('name', metavar='NAME', type=str,
+                        help="A name for the cluster.")
+    parsed, _ = parser.parse_known_args(args)
+
+    # Get the appdir path
+    appdir = safe_make_appdir("k8s-alpha")
+    terrapath = os.path.join(appdir, parsed.name)
+    # Move to the terraform directory
+    os.chdir(terrapath)
+    call_or_exit(['terraform', 'destroy'])
+
+    # TODO: Also delete all NodeBalancers and Volumes using the cluster prefix
+
+def quoted_string_or_bare_int(val):
+    if type(val) is int:
+        return val
+    elif type(val) is str:
+        return '"{}"'.format(val)
+    else:
+        return None
+
+def gen_terraform_file(context, cluster_name, hashname):
+    tf_file_parts = []
+
+    for varname in tf_var_map.keys():
+        tf_file_parts.append("""variable "{tf_varname}" {{
+  default = {default}
+}}
+""".format(tf_varname=tf_var_map[varname]['name'],
+           default=quoted_string_or_bare_int(tf_var_map[varname]['default'])))
+    
+    tf_file_parts.append("""module "k8s" {{
+  source  = "git@github.com:linode/terraform-linode-k8s.git?ref=for-cli"
+
+  linode_token = "{token}"
+  
+  linode_group = "ka{hashname}-{cluster_name}"
+""".format(token=context.token,
+           cluster_name=cluster_name,
+           hashname=hashname,))
+
+    for varname in tf_var_map.keys():
+        tf_file_parts.append("""
+  {tf_varname} = "${{var.{tf_varname}}}"
+""".format(tf_varname=tf_var_map[varname]['name']))
+
+    tf_file_parts.append("}\n")
+
+    return ''.join(tf_file_parts)
+
+def gen_terraform_args(parsed):
+    args = []
+    for varname in tf_var_map.keys():
+        args = args + ['-var', "{}={}".format(tf_var_map[varname]['name'], getattr(parsed, varname))]
+    return args
+
+def call_or_exit(*args, **kwargs):
+    ret = spcall(*args, **kwargs)
+    if ret != 0:
+        sys.exit(ret)
```

Comment:
I assume the called process would've printed an error in this case - is it worth printing something else anyway?